### PR TITLE
Allow timestamp with timezone on mssql databases

### PR DESCRIPTION
**Pull request by caiquecastro**
The SQL Server supports the following date / time data types:

date (Transact-SQL)
datetime (Transact-SQL)
datetime2 (Transact-SQL)
datetimeoffset (Transact-SQL)
smalldatetime (Transact-SQL)
time (Transact-SQL)

In my opinion, timestamp can be represented by (datetime, datetime2, datetimeoffset and smalldatetime).

Regarding the docs and integration tests, I want to wait for the reply about the incompatibility between dialects on the parameter for timestamp function on schema builder.

### CODE REVIEW

File: src/dialects/mssql/schema/columncompiler.js [hunk 1]
```diff
@@ -62,7 +62,9 @@ assign(ColumnCompiler_MSSQL.prototype, {
 
   datetime: 'datetime',
 
-  timestamp: 'datetime',
+  timestamp(useTz) {
```
Comment (elhigu) [inline src/dialects/mssql/schema/columncompiler.js]: use options object instead of single boolean parameter.

Comment (caiquecastro) [reply src/dialects/mssql/schema/columncompiler.js]: Regarding the object parameter on the function, it will be different between the dialects

**Comment by elhigu** [comment]
Code seems to be fine, this is so simple change that I don't require an integration test for it now... However documentation PR is still needed before this can be merged.

**Comment by caiquecastro** [comment]
You mean the PR on knex/documentation repo?

I just did it.

https://github.com/knex/documentation/pull/128

Please check for me.

Thanks

**Comment by elhigu** [comment]
Thanks!

**Pull request merged**